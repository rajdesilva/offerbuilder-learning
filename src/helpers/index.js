export * from './api-config';
export * from './history';
