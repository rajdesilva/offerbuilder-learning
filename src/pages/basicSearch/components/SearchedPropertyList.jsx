import React, { Fragment, useEffect, useState } from "react";
import {
  Col,
  Row,
  Rate,
  Table,
  Empty,
  Button,
  Skeleton,
  message,
  Spin,
  Divider,
  Space,
} from "antd";
import moment from "moment/min/moment-with-locales";
import { Link, useRouteMatch } from "react-router-dom";
import { useDispatch, useSelector } from "react-redux";
import defaultImg from "../../../assets/images/default-img.svg";
import copyIcon from "../../../assets/images/copy-icon.svg";
import {
  browseSupplyActions,
  propertyCartActions,
  supplySearchActions,
} from "../actions";
import { ConfirmAddCartItemModal } from "./ConfirmAddCartItemModal";
import {
  displayMargin,
  getFormattedValue,
  getMarketPriceForPropertyFromList,
  isParentOrChildrenPropertyAdded,
  isPropertySelected,
  getParsedProperty,
} from "../../../helpers/utility";
import { FormattedMessage, useIntl } from "react-intl";
import { appConstants } from "../../../common";
import { EmptyList } from "./EmptyList";
import { store } from "../../../redux/store";
import { isEqual } from "lodash";
import { searchSupply } from "../service";
import { newOfferActions } from "../../offers/actions";
import { LoadingOutlined } from "@ant-design/icons";
import styles from "./css/SearchedPropertyList.module.less";

import serviceGlobal from '../api-service';
import { advancedSearch } from '../service';

import SearchResultPagination from './SearchResultPagination';

import heartImg from '../../../assets/images/hrt.svg'
import noImg from '../../../assets/images/no-img.svg'
import star from '../../../assets/images/start.svg'

export function SearchedPropertyList({ prev, show, isEditFlow }) {
  const intl = useIntl();
  const {
    selectedProperties,
    loading,
    marketPriceLoading,
    includeMarketPrice,
  } = useSelector((state) => ({
    selectedProperties: window.getValue(state, "propertycart.cartItems"),
    loading: window.getValue(state, "searchedproperties.loading"),
    marketPriceLoading: window.getValue(state, "marketprice.loading"),
    includeMarketPrice: window.getValue(
      state,
      "marketprice.includeMarketPrice"
    ),
  }));
  let properties = useSelector((state) =>
    window.getValue(state, "searchedproperties.properties")
  );
  const currencyCode = useSelector(
    (state) => window.getValue(state, "searchparams.currencyCode"),
    isEqual
  );
  const lastSearchedDistance = useSelector(
    (state) => window.getValue(state, "searchparams.distance"),
    isEqual
  );
  const destination = useSelector(
    (state) => window.getValue(state, "searchparams.destination"),
    isEqual
  );
  const [modalInfo, setModalInfo] = useState({});
  const dispatch = useDispatch();
  const match = useRouteMatch({
    path: "/offers/create-new-offer/:step?",
    strict: true,
    sensitive: true,
  });
  const [expandedRows, setExpandedRows] = useState([]);


  useEffect(() => {
    dispatch({
      type: supplySearchActions.SUPPLY_SEARCH_UPDATE_BASIC_REDUX_STATE_ALL,
      payload: {},
    });
  }, [])


  const [htName, sethtName] = useState("");
  const [haddress, setaddress] = useState("");
  const [hzip, setzip] = useState("");
  const [hcity, setCity] = useState("");
  const [hdistance, setdistance] = useState("");
  const [hcountry, setcountry] = useState("");
  const [hlatitude, setlatitude] = useState("");
  const [hlongitude, setlongitude] = useState("");
  const [hgroupType, setgroupType] = useState("ALL");
  const [hsearchType, setsearchType] = useState("");
  const [hpage, setpage] = useState(0);
  const [hpagesize, setpageSize] = useState(10);
  const [sortCriteria, setsortCriteria] = useState("");
  const [sortOrder, setsortOrder] = useState("");

  const [resultvalue, setResultValue] = useState([]);
  const [loadspinner, setloadsppinner] = useState(false);
  const [pagin, setpagin] = useState(false);
  const [showftr, setftr] = useState(false);




  const searchParamsbasic = useSelector((state) =>
    state.searchParamsBasic
  );

  useEffect(() => {
    if (searchParamsbasic) {

      let params = {};
      let propertyType = "ALL";
      let pageSizes = 10;
      let srchType = "BASIC";
      let pageOff = 0;

      if (searchParamsbasic.Address) {
        params["address"] = searchParamsbasic.Address;
      }
      if (searchParamsbasic.distance) {
        params["distance"] = searchParamsbasic.distance;
      }
      if (searchParamsbasic.countryCode) {
        params["countryCode"] = searchParamsbasic.countryCode;
      }
      if (searchParamsbasic.hotelName) {
        params["hotelName"] = searchParamsbasic.hotelName;
      }
      if (searchParamsbasic.latitude) {
        params["latitude"] = searchParamsbasic.latitude;
      }
      if (searchParamsbasic.longitude) {
        params["longitude"] = searchParamsbasic.longitude;
      }
      if (searchParamsbasic.zip) {
        params["zip"] = searchParamsbasic.zip;
      }
      if (searchParamsbasic.city) {
        params["city"] = searchParamsbasic.city;
      }
      if (searchParamsbasic.propertyGroupType) {
        propertyType = searchParamsbasic.propertyGroupType;
      }
      if (searchParamsbasic.pageSize) {
        pageSizes = searchParamsbasic.pageSize;
      }
      if (searchParamsbasic.globalSearchType) {
        srchType = searchParamsbasic.globalSearchType;
      }
      // if (searchParamsbasic.pageOffset) {
      //   pageOff = searchParamsbasic.pageOffset;
      // }

      retrievList(params, propertyType, pageSizes, srchType)
    }
  }, [searchParamsbasic]);


  const retrievList = (params, prType, prSize, srType) => {
    if (Object.keys(params).length > 0) {
      setloadsppinner(true);
      setftr(false)
      advancedSearch(params, srType, hpage, prSize, sortCriteria, sortOrder, prType)
        .then((response) => {
          store.dispatch({
            type: browseSupplyActions.BROWSE_SUPPLY_SEARCH_LOADING,
            payload: false,
          });
          console.log(response);

          if (response) {
            response.result.hasNext ? setpagin(true) : setpagin(false)
          }
          if (response.success) {
            setftr(true)
          }

          setResultValue(response.result.result)
          setloadsppinner(false);
        })
        .catch((e) => {
          console.log(e);
          setloadsppinner(false);
        });
    }
  };

  return (
    <Fragment>
      {
        loadspinner
          ? <Skeleton loading={loadspinner} active></Skeleton>
          :
          <div className={styles.wrapW}>

                {resultvalue &&
                  resultvalue.map((item) => {
                    const { id, name, address } = item;
                    let supplyName = [];
                    let imgLink = [];
                    let isImage = false;
                    let rating = [];
                    let isRate = false;
                    let imgStar = [];
                    return (
                      <Fragment>
                        {
                          item.knownIds &&
                          item.knownIds.map((subitem) => {
                            supplyName.unshift(subitem.idScheme.split(','));
                            if (subitem.property && subitem.property.info && subitem.property.info.images) {
                              imgLink.unshift(subitem.property.info.images);

                              subitem.property.info.images
                                ? isImage = true
                                : isImage = false
                            }
                            if (subitem.property && subitem.property.info && subitem.property.info.award) {
                                subitem.property.info.award
                                ? isRate = true
                                : isRate = false

                                rating.unshift(subitem.property.info.award.value);

                                // for(let i =0; i<=0; rating){
                                //   imgStar.push(`<img src=${star}/>`)
                                // }
                            }
                          })
                        }
                        <div key={id} className={styles.trWrap}>
                          <div className={styles.imgwrap}>
                            {/* {console.log(name, imgLink)} */}
                            {isImage ? <img src={imgLink[0][0]} alt="" width="121" /> :
                              <div className={styles.noImg}>
                                <img src={noImg} alt="pic" />
                              </div>
                            }
                          </div>
                          <div className={styles.trp}>
                          <div className={styles.nm} style={{fontSize: '16px'}}>
                            <h >{name}</h>  <br />
                            <div className={styles.adr}style={{fontSize: '14px'}}>
                              {address}
                            </div>
                          </div>
                          <div>
                            <span className={styles.sr}style={{fontSize: '12px'}}>Start Ratings</span>  <br />
                            <div className={styles.adr}>
                              {
                                isRate 
                                ? 
                                <img src={star} alt=""/>
                                : 'N/A'
                              }
                             
                              </div>
                          </div>
                          <div className={styles.adw}>
                            <span className={styles.sr}style={{fontSize: '12px'}}>Suppliers</span>
                            <div className={styles.adr}style={{fontSize: '14px'}}>
                                {
                                  supplyName.slice(0, 2).join(", ")
                                }
                            </div>
                          </div>
                          <div>
                            <button className={styles.btn}>
                              <img src={heartImg} alt="Heart" />
                              Add To Wishlist</button>
                          </div>
                          </div>
                        </div>
                      </Fragment>
                    );
                  })}
          </div>
      }
      {
        showftr
          ?
          <div className={styles.pageWrap}>
            <div className={styles.selectWrap}>

              <span>Rows Per Page</span>
              <select
                onChange={(e) => {
                  console.log(e.target.value)
                  dispatch({
                    type: supplySearchActions.SUPPLY_SEARCH_UPDATE_PSIZE,
                    payload: e.target.value,
                  });
                }}
              >
                <option value="10">10</option>
                <option value="20">20</option>
                <option value="50">50</option>
                <option value="100">100</option>
              </select>
            </div>

            <div className={styles.pageNum}>
              <button
                disabled={hpage < 1}
                onClick={() => {
                  setpage(hpage - 1)
                  dispatch({
                    type: supplySearchActions.SUPPLY_SEARCH_UPDATE_PNUM,
                    payload: hpage,
                  });
                }}
              >
                <svg viewBox="64 64 896 896" focusable="false" data-icon="left" width="1em" height="1em" fill="currentColor" aria-hidden="true"><path d="M724 218.3V141c0-6.7-7.7-10.4-12.9-6.3L260.3 486.8a31.86 31.86 0 000 50.3l450.8 352.1c5.3 4.1 12.9.4 12.9-6.3v-77.3c0-4.9-2.3-9.6-6.1-12.6l-360-281 360-281.1c3.8-3 6.1-7.7 6.1-12.6z"></path></svg>
              </button>
              <div className={styles.textNum}>
                {hpage + 1}
              </div>
              <button
                disabled={!pagin}
                onClick={() => {
                  setpage(hpage + 1)
                  dispatch({
                    type: supplySearchActions.SUPPLY_SEARCH_UPDATE_PNUM,
                    payload: hpage,
                  });
                }}
              >
                <svg viewBox="64 64 896 896" focusable="false" data-icon="right" width="1em" height="1em" fill="currentColor" aria-hidden="true"><path d="M765.7 486.8L314.9 134.7A7.97 7.97 0 00302 141v77.3c0 4.9 2.3 9.6 6.1 12.6l360 281.1-360 281.1c-3.9 3-6.1 7.7-6.1 12.6V883c0 6.7 7.7 10.4 12.9 6.3l450.8-352.1a31.96 31.96 0 000-50.4z"></path></svg>
              </button>
            </div>
          </div>
          : ''
      }
    </Fragment>
  );
}
