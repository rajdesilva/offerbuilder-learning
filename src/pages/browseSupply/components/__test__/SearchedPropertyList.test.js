import React from "react";
import { createStore } from "redux";
import { render, fireEvent } from "../../../../helpers/testUtils";
import { createMemoryHistory } from "history";
import { Router, MemoryRouter } from "react-router-dom";
import { SearchedPropertyList } from "../SearchedPropertyList";
import { ReduxStoreForPropertySearch } from "./ReduxStoreForPropertySearch";

jest.mock("./../../service");
const history = createMemoryHistory();

describe("display property component test", () => {
  test("create display property components snapshot", () => {
    const store = createStore(() => ({ ...ReduxStoreForPropertySearch }));
    const wrapper = render(
      <Router history={history}>
        <SearchedPropertyList />
      </Router>,
      {
        store,
      }
    );
    const { container } = wrapper;
    expect(container.firstChild).toMatchSnapshot();
  });

  test("check data is available or not", () => {
    const store = createStore(() => ({ ...ReduxStoreForPropertySearch }));
    const { getByText } = render(
      <Router history={history}>
        <SearchedPropertyList />
      </Router>,
      {
        store,
      }
    );
    expect(getByText(/AJSG Hotel/i)).toBeInTheDocument(); //hotel name
    expect(getByText(/Best hotel/i)).toBeInTheDocument(); // description
  });
  test("check image is available or not", () => {
    const store = createStore(() => ({ ...ReduxStoreForPropertySearch }));
    const { getByTestId } = render(
      <MemoryRouter initialEntries={["/offers/create-new-offer/:step?"]}>
        <SearchedPropertyList />
      </MemoryRouter>,
      {
        store,
      }
    );
    const image = getByTestId("image");
    expect(image).toHaveAttribute(
      "src",
      "https://res.cloudinary.com/seekda-dev/image/upload/w_1920,h_1080,c_fill,f_auto,fl_lossy,q_auto/development/DEMO_JEEVAN/chambre_double_green_fullsize_ecodef-1140x760.jpg"
    );
  });
  test("Add Property into cart", () => {
    const store = createStore(() => ({ ...ReduxStoreForPropertySearch }));
    store.dispatch = jest.fn();
    const { getByTestId } = render(
      <MemoryRouter initialEntries={["/offers/create-new-offer/:step?"]}>
        <SearchedPropertyList />
      </MemoryRouter>,
      {
        store,
      }
    );
    const addBtn = getByTestId("add-to-offer-btn");
    fireEvent.click(addBtn);
    expect(addBtn).toBeTruthy();
    expect(store.dispatch).toHaveBeenCalledTimes(6);
  });
  test("Dialog should open on adding the same property into Cart.", () => {
    const ts = ReduxStoreForPropertySearch;
    ts.propertycart.cartItems.push({
      img: "https://res.cloudinary.com/seekda-dev/image/upload/w_1920,h_1080,c_fill,f_auto,fl_lossy,q_auto/development/DEMO_JEEVAN/chambre_double_green_fullsize_ecodef-1140x760.jpg",
      images: [
        "https://res.cloudinary.com/seekda-dev/image/upload/w_1920,h_1080,c_fill,f_auto,fl_lossy,q_auto/development/DEMO_JEEVAN/chambre_double_green_fullsize_ecodef-1140x760.jpg",
        "https://res.cloudinary.com/seekda-dev/image/upload/w_1920,h_1080,c_fill,f_auto,fl_lossy,q_auto/development/DEMO_JEEVAN/DSC00137-ch_simple2-e1511971588521.jpg",
      ],
      propertyCode: "DEMO_JEEVAN",
      lcn: false,
      supplier: "ntp",
      channel: "NEMO_CH_06",
      hotelName: "AJSG Hotel",
      remainingCapital: null,
      city: "Vienna",
      country: "Austria",
      rating: "",
      description:
        "Our luxurious Apartment hotel in country style is situated directly in the untouched nature of a side valley of the Vinschgau. At our house you can relax 24 hours a day: Start your day with a hearty breakfast and afterwards enjoy the beautiful nature and the various sport activities we have to offer. Relax in our alpine sauna world or our Panorama swimming pool. In the evening you can choose between three different restaurants in which we serve regional and international specialties.",
      descriptions: [
        {
          text: "Our luxurious Apartment hotel in country style is situated directly in the untouched nature of a side valley of the Vinschgau. At our house you can relax 24 hours a day: Start your day with a hearty breakfast and afterwards enjoy the beautiful nature and the various sport activities we have to offer. Relax in our alpine sauna world or our Panorama swimming pool. In the evening you can choose between three different restaurants in which we serve regional and international specialties.",
          lang: "EN",
        },
      ],
      trustyou: "",
      isParent: true,
      key: "DEMO_JEEVAN-0true",
    });
    const store = createStore(() => ({ ...ts }));
    store.dispatch = jest.fn();
    const { getByTestId, getByText } = render(
      <MemoryRouter initialEntries={["/offers/create-new-offer/:step?"]}>
        <SearchedPropertyList />
      </MemoryRouter>,
      {
        store,
      }
    );
    const addBtn = getByTestId("add-to-offer-btn");
    fireEvent.click(addBtn);
    expect(getByText("Property already added")).toBeInTheDocument();
  });
  test("Show the added Label on the searched property which are already in the cart", () => {
    const ts = ReduxStoreForPropertySearch;
    ts.propertycart.cartItems.push({
      img: "https://res.cloudinary.com/seekda-dev/image/upload/w_1920,h_1080,c_fill,f_auto,fl_lossy,q_auto/development/DEMO_JEEVAN/chambre_double_green_fullsize_ecodef-1140x760.jpg",
      images: [
        "https://res.cloudinary.com/seekda-dev/image/upload/w_1920,h_1080,c_fill,f_auto,fl_lossy,q_auto/development/DEMO_JEEVAN/chambre_double_green_fullsize_ecodef-1140x760.jpg",
        "https://res.cloudinary.com/seekda-dev/image/upload/w_1920,h_1080,c_fill,f_auto,fl_lossy,q_auto/development/DEMO_JEEVAN/DSC00137-ch_simple2-e1511971588521.jpg",
      ],
      propertyCode: "DEMO_JEEVAN",
      lcn: false,
      supplier: "ntp",
      channel: "NEMO_CH_04",
      hotelName: "AJSG Hotel",
      remainingCapital: null,
      city: "Vienna",
      country: "Austria",
      rating: "",
      description:
        "Our luxurious Apartment hotel in country style is situated directly in the untouched nature of a side valley of the Vinschgau. At our house you can relax 24 hours a day: Start your day with a hearty breakfast and afterwards enjoy the beautiful nature and the various sport activities we have to offer. Relax in our alpine sauna world or our Panorama swimming pool. In the evening you can choose between three different restaurants in which we serve regional and international specialties.",
      descriptions: [
        {
          text: "Our luxurious Apartment hotel in country style is situated directly in the untouched nature of a side valley of the Vinschgau. At our house you can relax 24 hours a day: Start your day with a hearty breakfast and afterwards enjoy the beautiful nature and the various sport activities we have to offer. Relax in our alpine sauna world or our Panorama swimming pool. In the evening you can choose between three different restaurants in which we serve regional and international specialties.",
          lang: "EN",
        },
      ],
      trustyou: "",
      isParent: true,
      key: "DEMO_JEEVAN-0true",
    });
    const store = createStore(() => ({ ...ts }));
    store.dispatch = jest.fn();
    const { getByText } = render(
      <MemoryRouter initialEntries={["/offers/create-new-offer/:step?"]}>
        <SearchedPropertyList />
      </MemoryRouter>,
      {
        store,
      }
    );
    expect(getByText("Added")).toBeInTheDocument();
  });
  test("Show the added Label on the searched property which are already in the cart, clicking on the added label should remove the property from Cart", () => {
    const ts = ReduxStoreForPropertySearch;
    ts.propertycart.cartItems.push({
      img: "https://res.cloudinary.com/seekda-dev/image/upload/w_1920,h_1080,c_fill,f_auto,fl_lossy,q_auto/development/DEMO_JEEVAN/chambre_double_green_fullsize_ecodef-1140x760.jpg",
      images: [
        "https://res.cloudinary.com/seekda-dev/image/upload/w_1920,h_1080,c_fill,f_auto,fl_lossy,q_auto/development/DEMO_JEEVAN/chambre_double_green_fullsize_ecodef-1140x760.jpg",
        "https://res.cloudinary.com/seekda-dev/image/upload/w_1920,h_1080,c_fill,f_auto,fl_lossy,q_auto/development/DEMO_JEEVAN/DSC00137-ch_simple2-e1511971588521.jpg",
      ],
      propertyCode: "DEMO_JEEVAN",
      lcn: false,
      supplier: "ntp",
      channel: "NEMO_CH_04",
      hotelName: "AJSG Hotel",
      remainingCapital: null,
      city: "Vienna",
      country: "Austria",
      rating: "",
      description:
        "Our luxurious Apartment hotel in country style is situated directly in the untouched nature of a side valley of the Vinschgau. At our house you can relax 24 hours a day: Start your day with a hearty breakfast and afterwards enjoy the beautiful nature and the various sport activities we have to offer. Relax in our alpine sauna world or our Panorama swimming pool. In the evening you can choose between three different restaurants in which we serve regional and international specialties.",
      descriptions: [
        {
          text: "Our luxurious Apartment hotel in country style is situated directly in the untouched nature of a side valley of the Vinschgau. At our house you can relax 24 hours a day: Start your day with a hearty breakfast and afterwards enjoy the beautiful nature and the various sport activities we have to offer. Relax in our alpine sauna world or our Panorama swimming pool. In the evening you can choose between three different restaurants in which we serve regional and international specialties.",
          lang: "EN",
        },
      ],
      trustyou: "",
      isParent: true,
      key: "DEMO_JEEVAN-0true",
    });
    const store = createStore(() => ({ ...ts }));
    store.dispatch = jest.fn();
    const { getByTestId } = render(
      <MemoryRouter initialEntries={["/offers/create-new-offer/:step?"]}>
        <SearchedPropertyList />
      </MemoryRouter>,
      {
        store,
      }
    );
    const addBtn = getByTestId("add-to-offer-btn");
    fireEvent.click(addBtn);
    expect(store.dispatch).toHaveBeenCalledTimes(3);
  });
  test("check duplicate expanded row is available or not", () => {
    const store = createStore(() => ({ ...ReduxStoreForPropertySearch }));
    const { getByText, getAllByText } = render(
      <MemoryRouter initialEntries={["/offers/create-new-offer/:step?"]}>
        <SearchedPropertyList />
      </MemoryRouter>,
      {
        store,
      }
    );
    const showDuplicateBtn = getByText("Show Duplicate Properties");
    expect(showDuplicateBtn).toBeTruthy();
    fireEvent.click(showDuplicateBtn);
    expect(getByText("Hide Duplicates")).toBeTruthy();
    expect(getAllByText("AJSG Hotel")).toBeTruthy();
  });
  test("SearchedPropertyList check if market price api is being called based on redux flag or not and loader is displayed", () => {
    ReduxStoreForPropertySearch.marketprice.marketPriceData = [
      {
        price: 475,
        trustYouId: "6c9991fe-e182-447b-8e87-dc261b3ab3f3",
        provider: "expedia",
      },
    ];
    ReduxStoreForPropertySearch.marketprice.includeMarketPrice = true;
    ReduxStoreForPropertySearch.marketprice.loading = true;
    ReduxStoreForPropertySearch.searchedproperties.properties = [
      {
        propertyCode: "DEMO_JEEVAN",
        mainProperty: {
          info: {
            name: "AJSG Hotel",
            description: "Best hotel",
            city: "Wien",
            countryName: "Austria",
            images: [
              "https://res.cloudinary.com/seekda-dev/image/upload/w_1920,h_1080,c_fill,f_auto,fl_lossy,q_auto/development/DEMO_JEEVAN/chambre_double_green_fullsize_ecodef-1140x760.jpg",
            ],
            trustyou: {
              id: "6c9991fe-e182-447b-8e87-dc261b3ab3f3",
              info: {
                name: "Bishop's Lodge Resort and Spa",
                score: "87",
                ty_id: "6c9991fe-e182-447b-8e87-dc261b3ab3f3",
                score_description: "Excellent",
                sources_count: 17,
                reviews_count: 15,
              },
            },
            award: null,
          },
          highestMargin: 11,
          lowestMargin: 11,
          propertyCode: "DEMO_JEEVAN",
          lcn: false,
          remainingCapital: null,
          supplier: "ntp",
          channel: "NEMO_CH_04",
        },
        alternateProperties: [
          {
            info: {
              name: "AJSG Hotel",
              description:
                "Our luxurious Apartment hotel in country style is situated directly in the untouched nature of a side valley of the Vinschgau. At our house you can relax 24 hours a day: Start your day with a hearty breakfast and afterwards enjoy the beautiful nature and the various sport activities we have to offer. Relax in our alpine sauna world or our Panorama swimming pool. In the evening you can choose between three different restaurants in which we serve regional and international specialties.",
              city: "Wien",
              countryName: "Austria",
              images: [
                "https://res.cloudinary.com/seekda-dev/image/upload/w_1920,h_1080,c_fill,f_auto,fl_lossy,q_auto/development/DEMO_JEEVAN/chambre_double_green_fullsize_ecodef-1140x760.jpg",
              ],
              trustyou: null,
              award: null,
            },
            propertyCode: "DEMO_JEEVAN",
            lcn: false,
            remainingCapital: null,
            supplier: "ntp",
            channel: "NEMO_CH_06",
          },
        ],
      },
    ];
    const store = createStore(() => ({ ...ReduxStoreForPropertySearch }));
    const { getByTestId } = render(
      <MemoryRouter initialEntries={["/offers/create-new-offer/:step?"]}>
        <SearchedPropertyList />
      </MemoryRouter>,
      {
        store,
      }
    );
    expect(
      getByTestId("6c9991fe-e182-447b-8e87-dc261b3ab3f3-market-price-loader")
    ).toBeTruthy();
  });

  test("SearchedPropertyList check if market price api is called and based on redux data, loader is hidden and market price is displayed", () => {
    ReduxStoreForPropertySearch.marketprice.marketPriceData = [
      {
        price: 475,
        trustYouId: "6c9991fe-e182-447b-8e87-dc261b3ab3f3",
        provider: "expedia",
      },
    ];
    ReduxStoreForPropertySearch.marketprice.includeMarketPrice = true;
    ReduxStoreForPropertySearch.marketprice.loading = false;
    ReduxStoreForPropertySearch.searchedproperties.properties = [
      {
        propertyCode: "DEMO_JEEVAN",
        mainProperty: {
          info: {
            name: "AJSG Hotel",
            description: "Best hotel",
            city: "Wien",
            countryName: "Austria",
            images: [
              "https://res.cloudinary.com/seekda-dev/image/upload/w_1920,h_1080,c_fill,f_auto,fl_lossy,q_auto/development/DEMO_JEEVAN/chambre_double_green_fullsize_ecodef-1140x760.jpg",
            ],
            trustyou: {
              id: "6c9991fe-e182-447b-8e87-dc261b3ab3f3",
              info: {
                name: "Bishop's Lodge Resort and Spa",
                score: "87",
                ty_id: "6c9991fe-e182-447b-8e87-dc261b3ab3f3",
                score_description: "Excellent",
                sources_count: 17,
                reviews_count: 15,
              },
            },
            award: null,
          },
          highestMargin: 11,
          lowestMargin: 11,
          propertyCode: "DEMO_JEEVAN",
          lcn: false,
          remainingCapital: null,
          supplier: "ntp",
          channel: "NEMO_CH_04",
        },
        alternateProperties: [
          {
            info: {
              name: "AJSG Hotel",
              description:
                "Our luxurious Apartment hotel in country style is situated directly in the untouched nature of a side valley of the Vinschgau. At our house you can relax 24 hours a day: Start your day with a hearty breakfast and afterwards enjoy the beautiful nature and the various sport activities we have to offer. Relax in our alpine sauna world or our Panorama swimming pool. In the evening you can choose between three different restaurants in which we serve regional and international specialties.",
              city: "Wien",
              countryName: "Austria",
              images: [
                "https://res.cloudinary.com/seekda-dev/image/upload/w_1920,h_1080,c_fill,f_auto,fl_lossy,q_auto/development/DEMO_JEEVAN/chambre_double_green_fullsize_ecodef-1140x760.jpg",
              ],
              trustyou: null,
              award: null,
            },
            propertyCode: "DEMO_JEEVAN",
            lcn: false,
            remainingCapital: null,
            supplier: "ntp",
            channel: "NEMO_CH_06",
          },
        ],
      },
    ];
    const store = createStore(() => ({ ...ReduxStoreForPropertySearch }));
    const { queryByTestId, getByTestId } = render(
      <MemoryRouter initialEntries={["/offers/create-new-offer/:step?"]}>
        <SearchedPropertyList />
      </MemoryRouter>,
      {
        store,
      }
    );
    expect(
      queryByTestId("6c9991fe-e182-447b-8e87-dc261b3ab3f3-market-price-loader")
    ).not.toBeInTheDocument();
    expect(
      getByTestId("6c9991fe-e182-447b-8e87-dc261b3ab3f3-market-price-value")
    ).toBeInTheDocument();
  });
});
