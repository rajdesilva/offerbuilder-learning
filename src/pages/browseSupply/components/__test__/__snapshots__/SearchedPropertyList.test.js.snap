// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`display property component test create display property components snapshot 1`] = `
<div
  class="ant-table-wrapper"
>
  <div
    class="ant-spin-nested-loading"
  >
    <div
      class="ant-spin-container"
    >
      <div
        class="ant-table"
        data-testid="property-list-table"
      >
        <div
          class="ant-table-container"
        >
          <div
            class="ant-table-content"
          >
            <table
              style="table-layout: auto;"
            >
              <colgroup />
              <tbody
                class="ant-table-tbody"
              >
                <tr
                  class="ant-table-row ant-table-row-level-0 tr-height"
                  data-row-key="DEMO_JEEVAN-0true"
                >
                  <td
                    class="ant-table-cell"
                  >
                    <div
                      class="ant-row"
                      style="margin-left: -8px; margin-right: -8px;"
                    >
                      <div
                        class="ant-col ant-col-6"
                        style="padding-left: 8px; padding-right: 8px;"
                      >
                        <div>
                          <img
                            alt=""
                            data-testid="image"
                            src="https://res.cloudinary.com/seekda-dev/image/upload/w_1920,h_1080,c_fill,f_auto,fl_lossy,q_auto/development/DEMO_JEEVAN/chambre_double_green_fullsize_ecodef-1140x760.jpg"
                          />
                        </div>
                      </div>
                      <div
                        class="ant-col ant-col-8"
                        style="padding-left: 8px; padding-right: 8px;"
                      >
                        <div
                          class="ant-space ant-space-horizontal ant-space-align-center"
                        >
                          <div
                            class="ant-space-item"
                          >
                            <a
                              href="/property/DEMO_JEEVAN"
                            >
                                
                              <span>
                                AJSG Hotel
                              </span>
                               
                               
                            </a>
                          </div>
                        </div>
                        <div>
                          Wien
                          , 
                          Austria
                            
                        </div>
                        <div>
                          Best hotel
                        </div>
                        <button
                          class="ant-btn ant-btn-link"
                          type="button"
                        >
                          <span
                            data-testid="alternative-property-count"
                          >
                            5
                          </span>
                          <span>
                              
                          </span>
                          <span
                            data-testid="show-duplicate-properties"
                          >
                            Show Duplicate Properties
                          </span>
                        </button>
                      </div>
                      <div
                        class="ant-col ant-col-10"
                        style="padding-left: 8px; padding-right: 8px;"
                      >
                        <div
                          class="ant-row"
                        >
                          <div
                            class="ant-col ant-col-24"
                          >
                            <div
                              class="ant-row"
                            />
                          </div>
                        </div>
                        <div
                          class="ant-row"
                        >
                          <div
                            class="ant-divider ant-divider-vertical"
                            role="separator"
                          />
                          <div
                            class="ant-col ant-col-11"
                          >
                            <div>
                              <div
                                class="ant-row"
                                style="margin: -4px -4px 4px -4px;"
                              >
                                <div
                                  class="ant-col ant-col-10"
                                  style="padding: 4px 4px 4px 4px;"
                                >
                                  Supplier
                                </div>
                                <div
                                  class="ant-col ant-col-14"
                                  style="padding: 4px 4px 4px 4px;"
                                >
                                  <strong>
                                    NTP
                                    -
                                    NEMO_CH_04
                                  </strong>
                                </div>
                              </div>
                              <div
                                class="ant-row"
                                style="margin: -4px -4px 4px -4px;"
                              >
                                <div
                                  class="ant-col ant-col-10"
                                  style="padding: 4px 4px 4px 4px;"
                                >
                                  Margin
                                </div>
                                <div
                                  class="ant-col ant-col-14"
                                  style="padding: 4px 4px 4px 4px;"
                                >
                                  <strong>
                                    11 %
                                  </strong>
                                </div>
                              </div>
                            </div>
                          </div>
                          <div
                            class="ant-divider ant-divider-vertical"
                            role="separator"
                          />
                          <div
                            class="ant-col ant-col-11"
                          >
                            <div>
                              <div>
                                <div>
                                  <div
                                    class="ant-row"
                                    style="margin: -4px -4px 4px -4px;"
                                  >
                                    <div
                                      class="ant-col ant-col-24"
                                      style="padding: 4px 4px 4px 4px;"
                                    >
                                      <div>
                                        Lowest Price On
                                         
                                        NA
                                      </div>
                                    </div>
                                  </div>
                                  <div
                                    class="ant-row"
                                    style="margin: -4px -4px 4px -4px;"
                                  >
                                    <div
                                      class="ant-col ant-col-11"
                                      style="padding: 4px 4px 4px 4px;"
                                    >
                                      <div>
                                        B2B Price
                                      </div>
                                    </div>
                                    <div
                                      class="ant-col ant-col-13"
                                      style="padding: 4px 4px 4px 4px;"
                                    >
                                      <div>
                                        <strong>
                                          NA
                                        </strong>
                                      </div>
                                    </div>
                                  </div>
                                  <div
                                    class="ant-row"
                                    style="margin: -4px -4px 4px -4px;"
                                  >
                                    <div
                                      class="ant-col ant-col-11"
                                      style="padding: 4px 4px 4px 4px;"
                                    >
                                      <div>
                                        B2C Price
                                      </div>
                                    </div>
                                    <div
                                      class="ant-col ant-col-13"
                                      style="padding: 4px 4px 4px 4px;"
                                    >
                                      <div>
                                        <strong>
                                          NA
                                        </strong>
                                      </div>
                                    </div>
                                  </div>
                                </div>
                              </div>
                            </div>
                          </div>
                        </div>
                      </div>
                    </div>
                  </td>
                </tr>
              </tbody>
            </table>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
